addons/mail/static/src/models/discuss/discuss.js

```diff
@@ -272,6 +272,23 @@ function factory(dependencies) {
             return [];
         }
 
+        /**
+         * @private
+         */
+        _onChangeThreadIsPinned() {
+            let thread = this.thread;
+            // No thread, or thread is being removed
+            // So we display discuss' default thread (usually the Inbox)
+            if (!thread || !thread.isPinned) {
+                const [model, id] = this.defaultInitActiveId.split('_');
+                thread = this.env.models['mail.thread'].find(thread =>
+                    thread.id === id && thread.model === model
+                );
+            }
+            if (thread && this.threadViewer && thread !== this.thread) {
+                this.threadViewer.update({ thread: [['link', thread]] });
+            }
```

Comment:
I would just either fallback to inbox or have thread viewer not linked to any thread.

`active id` is used to automatically open discuss with a provided thread, like on expand on chat window or during a page load (from URL). There are some strange management and fallbacks because could not know thread on page load (requires init_messaging) or thread could be simply wrong.

`defaultActiveId` could have be replaced by just fallback on opening mailbox inbox or have discuss open no thread at all. Need of this field is suspicious.